flask/launcher/src/main/java/net/corda/flask/launcher/Launcher.java

```diff
@@ -0,0 +1,127 @@
+package net.corda.flask.launcher;
+
+import lombok.SneakyThrows;
+import net.corda.flask.common.Flask;
+import net.corda.flask.common.ManifestEscape;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.InputStream;
+import java.lang.reflect.Constructor;
+import java.net.URI;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+public class Launcher {
+
+    private static final String CACHE_FOLDER_DEFAULT_NAME = "flask_cache";
+    private static final Logger log = LoggerFactory.getLogger(JarCache.class);
+
+    @SneakyThrows
+    private static Path getCurrentJar() {
+        String launcherClassName = Launcher.class.getName();
+        URL url = Launcher.class.getClassLoader().getResource(launcherClassName.replace('.', '/') + ".class");
+        if (url == null || !Objects.equals("jar", url.getProtocol()))
+            throw new IllegalStateException(String.format("The class %s must be used inside a JAR file", launcherClassName));
+        String path = url.getPath();
+        URI jarUri = new URI(path.substring(0, path.indexOf('!')));
+        return Paths.get(jarUri);
+    }
+
+    @SneakyThrows
+    public static void main(String[] args) {
+        Path currentJar = getCurrentJar();
+        Manifest manifest = new Manifest();
+        try(ZipFile jar = new ZipFile(currentJar.toFile())) {
+            ZipEntry manifestEntry = jar.getEntry(JarFile.MANIFEST_NAME);
+            jar.getInputStream(manifestEntry);
+            try (InputStream inputStream = jar.getInputStream(manifestEntry)) {
+                manifest.read(inputStream);
+            }
+        }
+        String mainClassName = manifest.getMainAttributes().getValue(Attributes.Name.MAIN_CLASS);
+        Class<? extends Launcher> launcherClass = (Class<? extends Launcher>) Class.forName(mainClassName);
```

Comment:
As a rule, it's A Good Idea :tm: also to provide the classloader to use, whenever Java allows. In this case:
```java
Class.forName(mainClassName, true, Launcher.class.getClassLoader());
```

Reply:
will do